app/Http/Controllers/Api/V1/SocialLoginController.php

```diff
@@ -207,7 +220,106 @@ private function handleLogin(Request $request, string $provider, string $baseRed
          */
     public function dtaCallback(Request $request, string $provider): mixed
     {
-        return $this->handleCallback($request, $provider, env('DTA_URL'), env('OPENATHENS_REDIRECT_DTA_URL'));
+        $openAthensRedirectUrl = env('OPENATHENS_REDIRECT_URL');
+
+        $user = null;
+        try {
+            if (strtolower($provider) === 'linkedin') {
+                $provider = 'linkedin-openid';
+            }
+            if (strtolower($provider) === 'openathens') {
+                $provider = 'open-athens';
+                if (session_status() === PHP_SESSION_NONE) {
+                    session_start();
+                }
+                $input = $request->all();
+                $code = array_key_exists('code', $input) ? $input['code'] : '';
+                $_REQUEST['code'] = $code;
+                $state = array_key_exists('state', $input) ? $input['state'] : '';
+                $_REQUEST['state'] = $state;
+                $_SESSION['openid_connect_state'] = $state;
+
+                $oidc = new OpenIDConnectClient(
+                    Config::get('services.openathens.issuer'),
+                    Config::get('services.openathens.client_id'),
+                    Config::get('services.openathens.client_secret')
+                );
+                $oidc->providerConfigParam([
+                    'authorization_endpoint' => env('OPENATHENS_ISSUER_URL') . '/oidc/auth',
+                    'jwks_uri' => env('OPENATHENS_ISSUER_URL') . '/oidc/jwks',
+                    'token_endpoint' => env('OPENATHENS_ISSUER_URL') . '/oidc/token',
+                    'userinfo_endpoint' => env('OPENATHENS_ISSUER_URL') . '/oidc/userinfo',
+                ]);
+
+                $oidc->setRedirectUrl($openAthensRedirectUrl);
+                $oidc->authenticate();
+
+                $response = $oidc->requestUserInfo();
+                $socialUser = json_decode(json_encode($response), true);
+                $socialUserDetails = $this->openathensResponse($socialUser, $provider);
+
+                $user = User::where('providerid', $socialUserDetails['providerid'])->first();
+            } else {
+                $providerURL = config("services.$provider.redirect");
```

Comment:
Always quicker and more performant to concat vars into strings. Rather than use magic quotes.